### PR TITLE
Added documentation for the json key new feature

### PR DESCRIPTION
**Pull request by jselvi**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs

This is part of a pull request modifying several repositories:
- https://github.com/returntocorp/semgrep/pull/6797
- https://github.com/returntocorp/semgrep-langs/pull/24
- https://github.com/returntocorp/semgrep-interfaces/pull/46
- https://github.com/returntocorp/semgrep-docs/pull/843

### CODE REVIEW

File: docs/writing-rules/experiments/extract-mode.md
Comment (adamkvitek) [inline docs/writing-rules/experiments/extract-mode.md:110]:
```suggestion
The `json` is an optional key in the extract mode. This boolean value defaults to `false` and specifies whether the extracted content is parsed as raw source code or as a JSON array. Setting the `json` key to `true` is useful in extracting code from JSON formats such as Jupyter Notebooks.
```

File: docs/writing-rules/experiments/extract-mode.md
Comment (adamkvitek) [inline docs/writing-rules/experiments/extract-mode.md:127]:
```suggestion
Although extract mode supports JSON array decoding with the `json` key, it does not support other additional processing for the extracted text, such as unescaping strings.
```

**Review by adamkvitek — COMMENTED** [review]
Thanks, leaving suggestions, please commit them or let me know with further comments. @jselvi

**Comment by jselvi** [comment]
I have committed the changes as suggested. Thanks @adamkvitek for your QA.

**Comment by adamkvitek** [comment]
@jselvi this feature is not out yet?

**Comment by jselvi** [comment]
@adamkvitek The new feature is in another pull request which is linked from this one. It has not been merged yet, no.

**Review by adamkvitek — APPROVED** [review]
Approving so you can merge this PR once related PRs are merged.

**Review by emjin — APPROVED** [review]
lgtm

File: docs/writing-rules/experiments/extract-mode.md
Comment (aryx) [inline docs/writing-rules/experiments/extract-mode.md:96]: here it's actually transform:
Thinking about it we should have called it  map, so we have a mapreduce inside extract mode

**Pull request merged**
